utils/emoji_utils.jsx

```diff
@@ -1,16 +1,50 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+const defaultRule = (aName, bName, emojiA, emojiB) => {
+    if (emojiA.category === 'custom' && emojiB.category !== 'custom') {
+        return 1;
+    } else if (emojiB.category === 'custom' && emojiA.category !== 'custom') {
+        return -1;
+    }
+
+    return aName.localeCompare(bName);
+};
+
+const thumbsDownRule = (other) =>
+    (other === 'thumbsup' || other === '+1' ? 1 : 0);
+const thumbsUpRule = (other) => (other === 'thumbsdown' || other === '-1' ? -1 : 0);
+
+const customRules = {
+    thumbsdown: thumbsDownRule,
+    '-1': thumbsDownRule,
+    thumbsup: thumbsUpRule,
+    '+1': thumbsUpRule,
+};
+
 export function compareEmojis(emojiA, emojiB, searchedName) {
-    const aName = emojiA.name || emojiA.aliases[0];
-    const bName = emojiB.name || emojiB.aliases[0];
+    // There's an edge case for custom emojis that start with a thumb prefix.
+    // It doesn't match the first alias for the relevant system emoji.
+    // We don't have control over the names or aliases of custom emojis...
+    // ... and how they compare to the relevant system ones.
+    // So we need to search for a matching alias in the whole array.
+    // E.g. thumbsup-custom vs [+1, thumbsup]
+
+    const getMatchingName = (emoji) =>
+        emoji.name || emoji.aliases.find((alias) => alias.startsWith(searchedName)) || emoji.aliases[0];
```

Comment:
I might be inclined to move this out to a separate function and maybe also add some validation for the argument given it's expected to be an object.

Reply:
Do you mean move it out of the current function scope? That way we'd do `(searchedName) => (emoji) => ...`, leading to a similar outcome.

Reply:
I just mean a standalone, non exported function with some added validation, something like this (haven't specifically tested):

``` javascript
const getEmojiName = (emoji, searchedName) => {
    if (!emoji || !(emoji.name || emoji.aliases) || !searchedName) {
        return '';
    }
    return emoji.name || emoji.aliases.find((alias) => alias.startsWith(searchedName)) || emoji.aliases[0];
};
```

Might be better to return null instead if validation here fails and add further validation after this function is used?

Reply:
I think returning a string here is ok because we don't check for truthiness. As far as I can see the validation here should cover all cases  👍